src/Evolution/Systems/Cce/Equations.hpp

```diff
@@ -0,0 +1,944 @@
+                          Tags::RegularIntegrand<Tags::BondiH>,
+                          Tags::LinearFactor<Tags::BondiH>,
+                          Tags::LinearFactorForConjugate<Tags::BondiH>>;
+};
+}  // namespace detail
+
+/// \brief A struct for providing a `tmpl::list` of integrand tags that need to
+/// be computed before integration can proceed for a given Bondi variable tag.
+template <typename BondiVariable>
+using integrand_terms_to_compute_for_bondi_variable =
+    typename detail::integrand_terms_to_compute_for_bondi_variable_impl<
+        BondiVariable>::type;
+
+/*!
+ * \brief Computes one of the inputs for the integration of one of the
+ * Characteristic hypersurface equations.
+ *
+ * \details The template argument must be one of the integrand-related prefix
+ * tags templated on a Bondi quantity tag for which that integrand is required.
+ * The relevant prefix tags are `Tags::Integrand`, `Tags::PoleOfIntegrand`,
+ * `Tags::RegularIntegrand`, `Tags::LinearFactor`, and
+ * `Tags::LinearFactorForConjugate`. The Bondi quantity tags that these tags may
+ * wrap are `Tags::BondiBeta`, `Tags::BondiQ`, `Tags::BondiU`, `Tags::BondiW`,
+ * and `Tags::BondiH`.
+ *
+ * The integrand terms which may be computed for a given Bondi variable are
+ * enumerated in the type alias `integrand_terms_to_compute_for_bondi_variable`,
+ * which takes as a single template argument the tag for which integrand terms
+ * would be computed, and is a `tmpl::list` of the integrand terms needed.
+ *
+ * The resulting quantity is returned by `not_null` pointer, and the required
+ * argument tags are given in `return_tags` and `argument_tags` type aliases,
+ * where the `return_tags` are passed by `not_null` pointer (so include
+ * temporary buffers) and the `argument_tags` are passed by const reference.
+ *
+ * Additional mathematical details for each of the computations can be found in
+ * the template specializations of this struct. All of the specializations have
+ * a static `apply` function which takes as arguments
+ * `SpinWeighted<ComplexDataVector, N>`s for each of the Bondi arguments.
+ */
+template <typename IntegrandTag>
+struct ComputeBondiIntegrand;
+
+/*!
+ * \brief Computes the integrand (right-hand side) of the equation which
+ * determines the radial (y) dependence of the Bondi quantity \f$\beta\f$.
+ *
+ * \details The quantity \f$\beta\f$ is defined via the Bondi form of the
+ * metric:
+ * \f[
+ * ds^2 = - \left(e^{2 \beta} (1 + r W) - r^2 h_{AB} U^A U^B\right) du^2 - 2
+ * e^{2 \beta} du dr - 2 r^2 h_{AB} U^B du dx^A + r^2 h_{A B} dx^A dx^B. \f]
+ * Additional quantities \f$J\f$ and \f$K\f$ are defined using a spherical
+ * angular dyad \f$q^A\f$:
+ * \f[ J \equiv h_{A B} q^A q^B, K \equiv h_{A B} q^A
+ * \bar{q}^B.\f]
+ * See \cite Bishop1997ik \cite Handmer2014qha for full details.
+ *
+ * In the compactified coordinate \f$ y \equiv 1 - 2 R/ r\f$, where \f$R\f$
+ * is determined by the worldtube initial data, the equation which
```

Comment:
R is a function of u, correct?  So I would change 'worldtube initial data' to 'worldtube data'.  (Or even better would be to be more explicit, "...where \f$r(u,\theta,\phi)\f$ is the Bondi radius of the \f$y=\f$const surface, and \f$R(u,\theta,\phi)\f$ is the Bondi radius of the worldtube").

 Ditto in docs in all the other functions.